docs/UPGRADE-4.0.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: page
+title: Upgrading from 3.x to 4.0
+---
+
+This document outlines the steps one should take when upgrading to the new major version.
+
+## Step 1: Review the new requirements
+
+Requirements have changed, this project now requires **PHP 7.4 or later** and **Moodle 3.8.3** or later.
+
+## Step 2: Review the change log
+
+Detailed information about what changed in Version 4 can be found in the [change log](CHANGELOG.md).
+
+## Step 3: Review the CIs configuration files
+
+### GitHub Actions
+
+Review the updated [gha.dist.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/master/gha.dist.yml)
+and update the `.github/workflows/ci.yml` (or whatever it's named) file in your plugin. For detailed information about the contents of that `ci.yml` file, please see this [help document](GHAFileExplained.md).
+
+A summary of the actions required, really simple for version 4, is:
+
+1. Change the `composer create-project` line and bump to from version 3 (^3) to version 4 (^4).
+2. Test it and, hopefully, party!
+
+### Travis CI
+
+Review the updated [.travis.dist.yml](https://github.com/moodlehq/moodle-plugin-ci/blob/master/.travis.dist.yml)
+and update your `.travis.yml` file in your plugin. For detailed information about the contents of `.travis.dist.yml`
+file, please see this [help document](TravisFileExplained.md).
```

Comment:
Using self-describing strings for links and avoiding using the same string for two different links are important accessibility criteria. 
```suggestion
file, please see our [sample Travis workflow file explanation](TravisFileExplained.md).
```